cyder/base/utils.py

```diff
@@ -36,39 +36,39 @@ def shell_out(command, use_shlex=True):
     return out, err, p.returncode
 
 
-def log(msg, log_level='LOG_DEBUG', to_syslog=False, to_stderr=True,
-        logger=syslog):
-    msg = unicode(msg)
-    if to_syslog:
-        ll = getattr(logger, log_level)
-        for line in msg.splitlines():
-            logger.syslog(ll, line)
-    if to_stderr:
-        stderr.write(msg + '\n')
-
-
-def run_command(command, command_logger=None, failure_logger=None,
-                failure_msg=None, ignore_failure=False):
-    if command_logger:
-        command_logger('Calling `{0}` in {1}'.format(command, os.getcwd()))
+class Logger(object):
+    """
+    A Logger logs messages passed to it. Possible destinations include stderr
+    and the system journal (syslog).
+    """
 
-    out, err, returncode = shell_out(command)
+    def log_debug(self, msg):
+        pass
+
+    def log_info(self, msg):
+        pass
+
+    def log_notice(self, msg):
+        pass
 
+    def error(self, msg):
+        raise Exception(msg)
+
+
+def run_command(command, logger=Logger(), ignore_failure=False,
```

Comment:
I forgot about that. In this, yeah, it doesn't matter. I guess I could add a comment noting that the same default Logger instance is shared between every call of the function.